File: backend/app/schemas/book.py
from typing import Optional
from pydantic import BaseModel, Field
from app.schemas.datetime_utils import UtcDatetime

class BookResponse(BaseModel):
    id: int
    title: str
    author: str
    isbn: Optional[str] = None
    cover_url: Optional[str] = None
    description: Optional[str] = None
    genre: Optional[str] = None
    language: str = "zh"
    created_at: UtcDatetime
    
    model_config = {"from_attributes": True}


class RecommendedTypeInfo(BaseModel):
    code: str
    name: str


class BookRatingRequest(BaseModel):
    """书籍评分请求（1-5 星）"""
    rating: int = Field(ge=1, le=5)


class BookRatingResponse(BaseModel):
    """书籍评分响应（DELETE 取消评分时 rating 为 None）"""
    rating: Optional[int] = None


class BookDetailResponse(BaseModel):
    id: int
    title: str
    author: str
    cover_url: Optional[str] = None
    description: Optional[str] = None
    genre: Optional[str] = None
    recommended_types: list[RecommendedTypeInfo] = []
    comment_count: int = 0
    is_favorited: bool = False
    # 评分汇总：avg_rating 无评分时为 None；my_rating 未登录或未评分为 None
    avg_rating: Optional[float] = None
    rating_count: int = 0
    my_rating: Optional[int] = None
    created_at: UtcDatetime

    model_config = {"from_attributes": True}

File: frontend/src/types/book.ts
// 书籍类型定义（对齐后端 schemas/book.py）
// BookResponse（列表/搜索摘要）与 BookDetailResponse（详情页）

export interface RecommendedTypeInfo {
    code: string
    name: string
}

// 对齐后端 BookResponse：列表/搜索卡片展示用
export interface BookSummary {
    id: number
    title: string
    author: string
    isbn?: string | null
    cover_url?: string | null
    description?: string | null
    genre?: string | null
    language: string
    created_at: string
}

// 对齐后端 BookDetailResponse：详情页用
export interface BookDetail extends BookSummary {
    recommended_types: RecommendedTypeInfo[]
    comment_count: number
    is_favorited: boolean
    avg_rating?: number | null
    rating_count: number
    my_rating?: number | null
}

// 搜索结果响应（对齐后端 ApiListResponse[BookResponse]）
export interface SearchResult {
    data: BookSummary[]
    total: number
    message: string
}
